cmd/honeyelb2/main.go

```diff
@@ -0,0 +1,249 @@
+        if err != nil {
+          // For some reason, we cannot write to
+          // the table or access it
+          logrus.WithField("tableName", state.DynamoTableName).Fatal("--highavail requires an existing DynamoDB table named appropriately, please refer to the README.")
+        }
+
+        stater = state.NewDynamoDBStater(sess, logbucket.AWSElasticLoadBalancing, opt.BackfillHr)
+        logrus.Info("State tracking with high availability enabled - using DynamoDB")
+
+      } else {
+        stater = state.NewFileStater(opt.StateDir, logbucket.AWSElasticLoadBalancing, opt.BackfillHr)
+        logrus.Info("State tracking enabled - using local file system.")
+      }
+      logrus.WithField("hours", time.Duration(opt.BackfillHr)*time.Hour).Debug("Backfill will be")
+
+      // TODO: Switch either publisher or parser?
+      // TODO: REMOVE
+      defaultPublisher := publisher.NewHoneycombPublisher(opt, stater, publisher.NewELBv2EventParser(opt.SampleRate))
+      downloadsCh := make(chan state.DownloadedObject)
+
+      // For now, just run one goroutine per-LB
+      for _, lbName := range lbNames {
+        logrus.WithFields(logrus.Fields{
+          "lbName": lbName,
+        }).Info("Attempting to ingest LB")
+
+        elbSvc := elbv2.New(sess, nil)
+
+        lbNameResp, err := elbSvc.DescribeLoadBalancers(&elbv2.DescribeLoadBalancersInput{
+          Names: []*string{
+           aws.String(lbName),
+          },
+        })
+        if err != nil {
+          fmt.Fprintln(os.Stderr, err)
+          os.Exit(1)
+        }
+
+        lbArn := lbNameResp.LoadBalancers[0].LoadBalancerArn
+        lbArnResp, err := elbSvc.DescribeLoadBalancerAttributes(&elbv2.DescribeLoadBalancerAttributesInput{
+          LoadBalancerArn: lbArn,
+        })
+        if err != nil {
+          fmt.Fprintln(os.Stderr, err)
+          os.Exit(1)
+        }
+
+        enabled := false
+        bucketName := ""
+        bucketPrefix := ""
+
+        for _, element := range lbArnResp.Attributes {
+          if *element.Key == "access_logs.s3.enabled" && *element.Value == "true" {
+            enabled = true
+          }
+          if *element.Key == "access_logs.s3.bucket" {
+            bucketName = *element.Value
+          }
+          if *element.Key == "access_logs.s3.prefix" {
+            bucketPrefix = *element.Value
+          }
+        }
+
+        if !enabled {
+          fmt.Fprintf(os.Stderr, `Access logs are not configured for ALB %q. Please enable them to use the ingest tool.
+
+For reference see this link:
+
+http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#enable-access-logging
+`, lbName)
+          os.Exit(1)
+        }
+        logrus.WithFields(logrus.Fields{
+          "bucket": bucketName,
+          "lbName": lbName,
+        }).Info("Access logs are enabled for ALB ♥")
+
+        elbDownloader := logbucket.NewELBDownloader(sess, bucketName, bucketPrefix, lbName, "alb")
```

Comment:
Perhaps change to `albDownloader`? Smaller change, but it makes what's happening here a bit more transparent.